### ISSUE TITLE
Замечания по задачам

### ISSUE THREAD

**Issue by EugenioAV** [avrudik/python_tasks#1]
**Задание 4.**

https://github.com/avrudik/python_tasks/blob/cfa3aeb6bfd7dd8f9c784062e7565d59dcd1f9f3/17.12.19/%D0%97%D0%B0%D0%B4%D0%B0%D0%BD%D0%B8%D0%B5%204.py#L4

Кажется, лишняя операция.

**Задание 6.**

В принципе, все правильно. Но решение сложное, тут используется и рекурсивная функция, и конструкция try-except. Можно было обойтись просто циклом while :)

**Задание 7.**

Еще будет полезно знать другой способ выполнения задачи на повышенную сложность - это очистить строку от пунктуации, прежде чем делить ее на отдельные слова с помощью метода split, а потом уже в списки искать самое длинное слово.

Очищение строки от лишних символов делается с помощью регулярных выражений.
Пример можно посмотреть здесь: https://www.pythonprogramming.in/how-to-remove-punctuation-marks-from-a-string-in-python.html

Прочитать про регулярные выражения в Python можно здесь: https://tproger.ru/translations/regular-expression-python/

**Comment by avrudik** [comment]
Да, с четвертым заданием полностью согласен, не помню уже, зачем эту строчку написал.
А вот 7-ю задачу я знал, как решить с использованием string, но решил, что так поинтересней будет, к тому же это заставило меня покопаться в юникоде и узнать чуточку больше, так что я не жалею, что выбрал именно такой способ) 
А вот про модуль "re" не знал, постараюсь взять на вооружение. Спасибо!
